inspect.go

```diff
@@ -0,0 +1,166 @@
+package enmime
+
+import (
+	"bufio"
+	"bytes"
+	"io"
+	"mime"
+	"net/textproto"
+	"strings"
+
+	"github.com/jhillyerd/enmime/internal/coding"
+	"github.com/pkg/errors"
+)
+
+func HumanHeadersOnly(b []byte) (map[string]string, error) {
+	b = clean(b)
+	tr := textproto.NewReader(bufio.NewReader(bytes.NewReader(b)))
+	headers, err := tr.ReadMIMEHeader()
+	switch errors.Cause(err) {
+	case nil, io.EOF:
+	// carry on, io.EOF is expected
+	default:
+		return nil, err
+	}
+	bs := bufio.NewScanner(bufio.NewReader(bytes.NewReader(b)))
+	res := map[string]string{}
+	for bs.Scan() {
+		line := bs.Text()
+		if strings.HasPrefix(line, " ") || strings.HasPrefix(line, "\t") {
+			continue
+		}
+		i := strings.Index(line, ":")
+		if i == -1 {
+			continue
+		}
+		header := textproto.CanonicalMIMEHeaderKey(line[:i])
+		if len(headers[header]) == 0 {
+			// somethings up, we should have already processed all of these, so why are we trying to fetch from an empty list, did we miscount?
+			continue
+		}
+		// pop
+		firstValue := headers[header][0]
+		// shift
+		headers[header] = headers[header][1:]
+
+		res[header] = rfc2047parts(firstValue)
+	}
+
+	return res, nil
+}
+
+func clean(b []byte) []byte {
+	slice := bytes.SplitAfter(b, []byte{'\r', '\n'})
+	dest := make([]byte, 0, len(b)+2)
+	headers := true
+	for _, v := range slice {
+		if (bytes.Index(v, []byte{':'}) > -1 || bytes.HasPrefix(v, []byte{' '}) || bytes.HasPrefix(v, []byte{'\t'})) && headers {
```

Comment:
Move headers to the front of the if clause to save indexing into byte slice unnecessarily.  Or we move this if block into the existing `if headers` block below to avoid testing it twice; if it doesn't make things unreadable.